R/TealAppDriver.R

```diff
@@ -0,0 +1,393 @@
+        if (length(dots) == 0) {
+          return(base_string)
+        }
+        (shiny::NS(base_string))(paste(dots, collapse = shiny::ns.sep))
+      }
+    },
+    #' @description
+    #' Get the input from the module in the `teal` app.
+    #' This function will only access inputs from the name space of the current active teal module.
+    #'
+    #' @param input_id (character) The shiny input id to get the value from.
+    #'
+    #' @return The value of the shiny input.
+    get_module_input = function(input_id) {
+      self$app$get_value(input = sprintf("%s-%s", self$get_active_ns("module"), input_id))
+    },
+    #' @description
+    #' Get the output from the module in the `teal` app.
+    #' This function will only access outputs from the name space of the current active teal module.
+    #'
+    #' @param output_id (character) The shiny output id to get the value from.
+    #'
+    #' @return The value of the shiny output.
+    get_module_output = function(output_id) {
+      self$app$get_value(output = sprintf("%s-%s", self$get_active_ns("module"), output_id))
+    },
+    #' @description
+    #' Set the input in the module in the `teal` app.
+    #' This function will only set inputs in the name space of the current active teal module.
+    #'
+    #' @param input_id (character) The shiny input id to get the value from.
+    #' @param value The value to set the input to.
+    #'
+    #' @return The `TealAppDriver` object invisibly.
+    set_module_input = function(input_id, value) {
+      self$app$set_inputs(
+        !!sprintf("%s-%s", self$get_active_ns("module"), input_id) := value
+      )
+      invisible(self$app)
+    },
+    #' @description
+    #' Get the active datasets that can be accessed via the filter panel of the current active teal module.
+    get_active_filter_vars = function() {
+      displayed_datasets_index <- sapply(
+        self$app$get_html(
+          sprintf(
+            "#%s-active-filter_active_vars_contents > span",
+            self$get_active_ns("filter_panel")
+          )
+        ),
+        function(x) {
+          style <- x %>%
+            rvest::read_html() %>%
+            rvest::html_node("span") %>%
+            rvest::html_attr("style")
+          style <- ifelse(is.na(style), "", style)
+          style != "display: none;"
+        }
+      ) %>%
+        unname()
+      available_datasets <- self$app$get_html(
+        sprintf(
+          "#%s-active-filter_active_vars_contents",
+          self$get_active_ns("filter_panel")
+        )
+      ) %>%
+        read_html() %>%
+        html_nodes(".filter_panel_dataname") %>%
+        html_text()
+      available_datasets[displayed_datasets_index]
+    },
+    #' @description
+    #' Get the active filter variables from a dataset in the `teal` app.
+    #'
+    #' @param dataset_name (character) The name of the dataset to get the filter variables from.
+    get_active_data_filters = function(dataset_name) {
```

Comment:
maybe get_active_filters should return a list:
- dataname
- varname
- selected

Then it will be easier in tests to compare expectations (less code to write)

Reply:
Nice idea, I agree.

Reply:
Extended `get_active_data_filters` to handle the case when `dataset_name` is `NULL`. When it is NULL it will return a list of all the active filter vars for all the datanames.

Reply:
I took a jab at this on #1161 , it still holds on to `get_active_filter_selection` as a private method.

Reply:
Tracking this on that PR. So, resolving.